### PR TITLE
Add bash file to install cro_lan framework easily

### PR DESCRIPTION
**Pull request by hhr293**

### CODE REVIEW

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 1]
```diff
@@ -0,0 +1,164 @@
+function version_lt() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" != "$1"; }
```
Comment (RodgerZhu) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:1]: will it be better to add below line at the first line?

#!/bin/bash

Comment (RodgerZhu) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:1]: Please remove the space in the file name.
install cr_lan framework .bash -> install cr_lan framework.bash

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+apt-get install -y build-essential python
+apt-get install -y autoconf gawk bison
+apt-get install -y libprotobuf-c-dev protobuf-c-compiler
+apt-get install -y python3-protobuf
+apt-get install -y unzip
+if [[ -d "/opt/intesl/sgxsdk/" ]]; then
+  echo "SGX SDK has been installed"
+elif [[   $os_version == *"20.04"* ]]; then
+  echo "begin to download SGX SDK"
+  wget https://download.01.org/intel-sgx/latest/dcap-latest/linux/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.17.100.3.bin
```
Comment (RodgerZhu) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:59]: The potential issue will be that if there is one new version SDK, the SDK version will not match so as to download failed here.

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+  wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+  echo "SGX SDK has been downloaded"
+elif [[  $os_version == *"18.04"* ]]; then
+  echo "begin to download SGX SDK"
+  wget https://download.01.org/intel-sgx/latest/dcap-latest/linux/distro/ubuntu18.04-server/sgx_linux_x64_sdk_2.17.100.3.bin
```
Comment (RodgerZhu) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:65]: Same as above.

**Review by RodgerZhu — COMMENTED** [review]
Please update the comments.
Other looks fine.

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        echo "Cross_language_framework can only be used in Ubuntu"
+        exit 1
+fi
+os_version=$(cat /etc/issue| sed '1!d' | sed -e 's/"//g' | awk '{print $2}')
+if version_lt $os_version $os_version_base; then
+        echo "Cross_language_framework can only be used in Ubuntu with version>=18.04"
+        exit 1
+fi
+gcc_version=$(gcc --version |sed '1!d' | sed -e 's/"//g' | awk '{print $4}')
+echo $gcc_version
```
Comment (liangintel) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:37]: please well add human readable string: "gcc version="

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+#install SGX SDK
+echo " begin to install SGX SDK Prerequisites"
+apt-get update
+apt-get install -y libssl-dev libcurl4-openssl-dev libprotobuf-dev
+apt-get install -y build-essential python
+apt-get install -y autoconf gawk bison
+apt-get install -y libprotobuf-c-dev protobuf-c-compiler
+apt-get install -y python3-protobuf
+apt-get install -y unzip
+if [[ -d "/opt/intesl/sgxsdk/" ]]; then
```
Comment (liangintel) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:55]: intesl -> intel

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+else
+  echo " No suitable SGX SDK version for this Ubuntu version"
+fi
+echo "begin to install SGX SDK"
+chmod +x *sdk*.bin
+./*sdk*.bin << EOF
+no
+/opt/intel/
+EOF
+echo " SGX SDK has been installed "
```
Comment (liangintel) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:78]: Need to check whether SDK has been successfully installed or not

File: cczoo/cross_lang_framework/install cr_lan framework .bash [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+sudo apt-get update
+sudo apt-get install -y libsgx-urts libsgx-dcap-ql libsgx-dcap-default-qpl
+sudo apt-get install -y libsgx-dcap-ql-dev libsgx-quote-ex-dev libsgx-dcap-quote-verify-dev
+# install SGX PCCS
+curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
+sudo apt-get install -y nodejs
+sudo apt-get install -y libcrack2    
+echo "begin to download SGX PCCS,you may need to give some input"            
+sudo apt-get install sgx-dcap-pccs 
+echo "you should configure PCCS by yourself"
```
Comment (liangintel) [inline cczoo/cross_lang_framework/install cr_lan framework .bash:90]: indicate how to configure pccs. e.g. specify the path of pccs configuration file

**Review by liangintel — APPROVED** [review]

**Pull request merged**